src/main/java/org/primeframework/mvc/util/CookieTools.java

```diff
@@ -70,7 +90,9 @@ public static <T> T fromCookie(String value, boolean encryptedIfOld, Encryptor e
     boolean compress = (result[3] & 0x02) == 0x02; // Second bit is compressed
     result = Arrays.copyOfRange(result, 4, result.length);
 
-    if (encrypt) {
+    if (encryptionRequired && !encrypt) {
+      throw new EncryptionException("Encryption is required to decrypt cookie but a non-encrypted cookie was presented");
```

Comment:
We should use complete sentences with punctuation.

Reply:
Done